### ISSUE TITLE
Fione Prediction Results for external application Question

### ISSUE THREAD

**Issue by gabfields02** [codelibs/docker-fione#3]
Good day.

Can i ask if there is a way to use the Fione Prediction Results to an external application or a web server like Javascript or Python Configuration?
Can the POJO be used for this? Or is there also another way?
If yes, can i also ask if there is a reference guide/manual I can look into for this?

**Comment by marevol** [comment]
Yes, but no reference in English yet... (Japanese is [here](https://fione.codelibs.org/docs/ja/getting-started/quickstart#docker%E3%81%A7%E4%BA%88%E6%B8%ACapi))
Fione provides Dockerfile with POJO model to serve REST API. It's Fione Serving server.
You can get it at a model details page.

**Comment by gabfields02** [comment]
Thank you, @marevol

**Comment by gabfields02** [comment]
Hi @marevol 

I am running Fione on a remote Linux server. 
I tried Fione Serving server and was able to query the model using the predict API with the curl command on the terminal.
However, I cannot access port 8081 on the remote server.
I want to ask if there is a user interface for this?

Also, is there a documentation on how to use the MOJO?

**Comment by marevol** [comment]
> I want to ask if there is a user interface for this?

I could not understand that... could you provide more info?

MOJO is a feature of H2O. Please see [H2O](http://docs.h2o.ai/h2o/latest-stable/h2o-docs/productionizing.html).

**Comment by gabfields02** [comment]
> MOJO is a feature of H2O. Please see H2O.
>
Thank you for this.


> I could not understand that... could you provide more info?
> 
![image](https://user-images.githubusercontent.com/55271708/86546553-16e79080-bf68-11ea-99d4-76b7ea38055c.png)
It says that Prediction API is on localhost:8081. However, I am running Fione on a remote server. 
I can do the queries in the terminal but I cannot access port 8081 on the browser. 
My assumption is that there is a user interface on port 8081. Is this correct?
Do you have an idea why port 8081 is not accessible for me on the browser?

**Comment by marevol** [comment]
Fione is not Prediction API server. 8081 is provided by a docker container for Prediction API.
So, if you run `docker run ...` command, you can access localhost:8081.

**Comment by gabfields02** [comment]
Thank you, @marevol. I got the Prediction working with java using the MOJO.

However, I have another question regarding these graphs.

![image](https://user-images.githubusercontent.com/55271708/86771988-bf3f4700-c085-11ea-8326-1ce499182440.png)

I am planning to display these graphs and the other tables shown in the results as well.
Can these results data be also integrated in my own development environment?
Is it still with the use of MOJO? If so, are there more documentations that might be useful?

**Comment by marevol** [comment]
Information of these graphs is provided by [H2O](https://docs.h2o.ai/h2o/latest-stable/h2o-docs/rest-api-reference.html).

**Comment by gabfields02** [comment]
Hello @marevol. This is my main function when integrating MOJO. Prediction in my own environment is working.

![image](https://user-images.githubusercontent.com/55271708/86896974-65985480-c139-11ea-887c-e0898d584137.png)


However, I'm having a hard time how to retrieve the results data (Scoring History, Variable Importances, etc) from the MOJO.
What is the way to do this? Is there a sample reference on how to do this?

**Comment by marevol** [comment]
For h2o-3, it's better to ask h2o-3 community.

**Comment by gabfields02** [comment]
> For h2o-3, it's better to ask h2o-3 community.

Okay. Thank you, @marevol

**Comment by gabfields02** [comment]
Hi, @marevol 

I'm thinking the solution to my problem is by using Binary Models.
![image](https://user-images.githubusercontent.com/55271708/87621813-3a72be00-c754-11ea-8c70-303d8d1a2a34.png)

I want to ask.  Instead of the MOJO, is there a way to download Binary Models from Fione?

**Comment by marevol** [comment]
Hmm, what is your use case to do that? I'm not sure about it...

**Comment by gabfields02** [comment]
I want to display training information/results like the Scoring History and Variable Importances. 
It is mentioned from this guide's video that there are some data in Binary Models (Binary File) that is not included in MOJOs.
https://www.coursera.org/lecture/machine-learning-h2o/pojo-and-mojo-SCCXv

![H2O binary models](https://user-images.githubusercontent.com/55271708/87631356-bdeada00-c769-11ea-8316-a58221f724f4.PNG)

Maybe I can find the Scoring History, Variable Importances, etc in the Binary Model.
My assumption is that Binary Model can also be retrieved from Fione. Is my assumption correct?

**Comment by marevol** [comment]
Fione provides these information in Model page.

**Comment by gabfields02** [comment]
Thank you. I also found them there as well.

However, what I want to achieve is display those information on my own in my own server/environment.
Would this be possible?

**Comment by marevol** [comment]
We have no plan to do that at the moment...

**Comment by gabfields02** [comment]
@marevol Okay, thank you.
